mlflow/types/schema.py

```diff
@@ -645,23 +658,52 @@ def _merge(self, map_type: "Map") -> "Map":
                 f"{map_type} with value_type={map_type.value_type}"
             )
 
-        if (
-            isinstance(self.value_type, (Array, Object, Map))
-            and self.value_type.__class__ is map_type.value_type.__class__
-        ):
+        if isinstance(self.value_type, (Array, Object, Map, AnyType)):
             return Map(value_type=self.value_type._merge(map_type.value_type))
 
         raise MlflowException(f"Map type {self!r} and {map_type!r} are incompatible")
 
 
+class AnyType:
+    def __init__(self):
+        """
+        AnyType can store any json-serializable data including None values.
+
+        .. Note::
+            AnyType should not be used to host union types in a schema.
+            It should only be used when the field is None, the type is not known
+            at the time of data creation, or the field can have multiple types.
+            e.g. for GenAI flavors, the model output could contain `None` values,
+            and `AnyType` can be used to represent them.
+        """
+
+    def __repr__(self) -> str:
+        return "Any"
+
+    def __eq__(self, other) -> bool:
+        return isinstance(other, AnyType)
+
+    def to_dict(self):
+        return {"type": ANY_TYPE}
+
+    def _merge(
+        self, another_type: Union["AnyType", Array, Object, Map, DataType]
```

Comment:
Can we define a base type for these types? Updating similar type list all over the place seems to be tedious when we add/remove a new type class.

Reply:
Make sense!